app/models/character.rb

```diff
@@ -26,9 +26,20 @@ class Character < ApplicationRecord
   validates :morality, numericality: { only_integer: true, greater_than_or_equal_to: 0,
                                        less_than_or_equal_to: 100 }
 
-  def assign_skills
+  def assign_skills(skills_names)
     JsonSkill.skills.each do |json_skill|
       skills.create(json_skill.attributes)
     end
+    upgrade_skills(skills_names)
```

Comment:
I'm not a fan of this in here...the `assign_skills` method was good, it had only one responsibility and now, it has two...